### PR TITLE
Script support local deployment in MacOS

### LINKED ISSUE #429: enrich the build scripts: support deploy local on mac
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by acezen**
## What do these changes do?

Script support local deployment in MacOS

## Related issue number

Fixes #429 

- [x] run gae
- [x] run gie
- [x] update document

### CODE REVIEW

File: docs/developer_guide.rst [hunk 1]
```diff
@@ -110,10 +110,23 @@ To make sure the maximum compatibility you may need:
 Note that if you want to build wheel packages for different Python versions, you may need to install multiple
 version of Python using `conda` or `pyenv`.
 
+The GraphScope analytical engine could be build on local on Mac with script.
```
Comment (siyuan0322) [inline docs/developer_guide.rst:113]: be build locally on mac

File: docs/developer_guide.rst [hunk 2]
```diff
@@ -110,10 +110,23 @@ To make sure the maximum compatibility you may need:
...
+
+If GraphScope's dependencies not satisfied，you could use install_denpendencies
```
Comment (siyuan0322) [inline docs/developer_guide.rst:115]: are not satisfied, you could use the

File: scripts/build.sh [hunk 3]
```diff
@@ -44,8 +44,8 @@ function check_os_compatibility() {
     exit 1
   fi
 
-  if [[ "${platform}" != *"Ubuntu"* ]]; then
-    echo "This script is only available on Ubuntu"
+  if [[ "${platform}" != *"Ubuntu"* && "${platform}" != *"Darwin"* ]]; then
+    echo "This script is only available on Ubuntu or MacOs"
```
Comment (siyuan0322) [inline scripts/build.sh:48]: MacOS

File: docs/developer_guide.rst [hunk 4]
```diff
@@ -110,10 +110,23 @@ To make sure the maximum compatibility you may need:
...
+script to install dependencies.
+
+.. code::shell
+    ./script/install_denpendencies.sh
+
+build the analytical engine with script on Mac.
```
Comment (siyuan0322) [inline docs/developer_guide.rst:121]: remove `on Mac`.

File: scripts/build.sh [hunk 5]
```diff
@@ -72,16 +68,33 @@ function check_dependencies_version() {
     echo "GraphScope requires python 3.6 or greater."
     exit 1
   fi
-  if ! command -v cargo &> /dev/null; then
+  if [[ "${platform}" != *"Darwin"* ]]; then
```
Comment (siyuan0322) [inline scripts/build.sh:71]: Why mac doesn't need the maven

Comment (siyuan0322) [reply scripts/build.sh:71]: and cargo

Comment (siyuan0322) [reply scripts/build.sh:71]: and go

File: scripts/build.sh [hunk 6]
```diff
@@ -120,7 +133,7 @@ function build_graphscope_gae() {
   cd ${graphscope_src}
   mkdir analytical_engine/build && pushd analytical_engine/build
   cmake ..
-  make -j`nproc`
+  sudo make -j`nproc`
```
Comment (siyuan0322) [inline scripts/build.sh:136]: Why make requires sudo

File: scripts/install_dependencies.sh [hunk 7]
```diff
@@ -45,8 +45,8 @@ function check_os_compatibility() {
     exit 1
   fi
 
-  if [[ "${platform}" != *"Ubuntu"* ]]; then
-    echo "This script is only available on Ubuntu"
+  if [[ "${platform}" != *"Ubuntu"* && "${platform}" != *"Darwin"* ]]; then
+    echo "This script is only available on Ubuntu or MacOs"
```
Comment (siyuan0322) [inline scripts/install_dependencies.sh:49]: MacOS

File: scripts/install_dependencies.sh [hunk 8]
```diff
@@ -127,6 +134,30 @@ function install_dependencies() {
...
+
+    brew install cmake double-conversion etcd protobuf apache-arrow openmpi boost glog gflags \
+      zstd snappy lz4 openssl@1.1 libevent fmt clang autoconf go maven
+
+    export OPENSSL_ROOT_DIR="/usr/local/opt/openssl"
+    export OPENSSL_LIBRARIES="/usr/local/opt/openssl/lib"
+    export OPENSSL_SSL_LIBRARY="/usr/local/opt/openssl/lib/libssl.dylib"
+
+    # install folly
+    wget https://github.com/facebook/folly/archive/v2020.10.19.00.tar.gz -P /tmp
```
Comment (siyuan0322) [inline scripts/install_dependencies.sh:148]: Why not use folly formulae in brew instead build from scratch.

Comment (acezen) [reply scripts/install_dependencies.sh:148]: brew install folly has problem to build vineyard.

File: scripts/build.sh (hunk 5 above)
Comment (acezen) [reply scripts/build.sh:71]: since gie is not ready yet in mac. Will remove these skip when gie ready.

File: docs/developer_guide.rst [hunk 9]
```diff
@@ -1,23 +1,23 @@
...
 
 Building and Testing
 --------------------
 
-GraphScope has many dependencies. 
+GraphScope has many dependencies.
 
-To make life easier, we provide two docker images with all needed dependencies 
-installed. 
+To make life easier, we provide two docker images with all needed dependencies
```
Comment (sighingnow) [inline docs/developer_guide.rst:14]: "needed" -> required.

File: docs/developer_guide.rst [hunk 10]
```diff
@@ -110,10 +110,23 @@ To make sure the maximum compatibility you may need:
 Note that if you want to build wheel packages for different Python versions, you may need to install multiple
 version of Python using `conda` or `pyenv`.
 
+The GraphScope analytical engine could be build on local on Mac with script.
+
+If GraphScope's dependencies not satisfied，you could use install_denpendencies
+script to install dependencies.
+
+.. code::shell
```
Comment (sighingnow) [inline docs/developer_guide.rst:118]: Leave a blank line here.

File: docs/developer_guide.rst [hunk 11]
```diff
@@ -110,10 +110,23 @@ To make sure the maximum compatibility you may need:
...
+    ./script/install_denpendencies.sh
+
+build the analytical engine with script on Mac.
+.. code::shell
```
Comment (sighingnow) [inline docs/developer_guide.rst:122]: Leave a blank line here.

File: scripts/install_dependencies.sh [hunk 12]
```diff
@@ -55,6 +55,13 @@ function check_os_compatibility() {
     exit 1
   fi
 
+  if [[ "${platform}" == *"Darwin"* ]]; then
+    if ! hash brew; then
+      echo "Homebrew not installed. Install Homebrew: https://docs.brew.sh/Installation."
```
Comment (sighingnow) [inline scripts/install_dependencies.sh:60]: Homebrew "is" not installed.

File: scripts/install_dependencies.sh [hunk 13]
```diff
@@ -127,6 +134,30 @@ function install_dependencies() {
     pip3 install libclang parsec setuptools wheel twine --user
   fi
 
+  if [[ "${platform}" == *"Darwin"* ]]; then
+    git config --global http.postBuffer 1048576000
```
Comment (sighingnow) [inline scripts/install_dependencies.sh:138]: Seems we don't use `git` here?

**Review by sighingnow — APPROVED** [review]
LGTM.

**Pull request merged**